src/Domain/Zone.php

```diff
@@ -2,33 +2,42 @@
 
 namespace SandwaveIo\RealtimeRegister\Domain;
 
+use SandwaveIo\RealtimeRegister\Domain\Enum\DomainZoneServiceEnum;
+
 final class Zone implements DomainObjectInterface
 {
-    public string $template;
+    public string $service;
+
+    public ?string $template;
 
     public ?bool $link;
 
-    private function __construct(string $template, ?bool $link)
+    private function __construct(string $service, ?string $template, ?bool $link)
     {
+        $this->service = $service;
         $this->template = $template;
         $this->link = $link;
     }
 
-    public static function fromArray(array $data): Zone
+    public static function fromArray(array $json): Zone
     {
-        return new Zone(
-            $data['template'],
-            $data['link'] ?? null
+        DomainZoneServiceEnum::validate($json['service']);
+
+        return new self(
+            $json['service'],
+            $json['template'] ?? null,
+            $json['link'] ?? null
         );
     }
 
     public function toArray(): array
     {
         return array_filter([
+            'service' => $this->service,
             'template' => $this->template,
             'link' => $this->link,
-        ], function ($x) {
-            return ! is_null($x);
+        ], static function ($x) {
```

Comment:
Why does this closure need to be static?

Reply:
it doesn't need to be per se, performance wise it's a micro-optimization. 

I have this extensions in my php storm (https://github.com/hammer-tools/php-hammer/wiki/Inspections#anonymous-function-can-be-static) and I simply don't like the warnings in my face :-)